lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -1027,6 +1058,9 @@ struct FuncOpConversion : public calyx::FuncOpPartialLoweringPattern {
     auto compOp = rewriter.create<calyx::ComponentOp>(
         funcOp.getLoc(), rewriter.getStringAttr(funcOp.getSymName()), ports);
 
+    std::string funcName = "func_" + funcOp.getSymName().str();
```

Comment:
See my other comment, I'm not sure you need to add the `func_` prefix.

Also, it is not safe to directly mutate operations in a rewrite pattern like this: https://mlir.llvm.org/docs/PatternRewriter/#restrictions. If possible, I would recommend just not using the prefix. But, if it is required for some reason, you need to do something like
```
rewriter.updateRootInPlace(funcOp, [](){ funcOp.setSymName(funcName); });
```

Reply:
@linuxlonelyeagle I think to be safe you should make this update, to use the updateRootInPlace. It might not seem an issue in the current tests, but if you don't use updateRootInPlace it can corrupt the IR in general.

Reply:
I didn't realize the message was answered so quickly, I've made the changes. Thank you very much for your suggestions.